interface/src/app/download/download.js

```diff
@@ -1,80 +1,74 @@
-  function DownloadController($scope, $window, AnnotationTypes) {
-    $scope.annotations = {
-      query_message: 'Connecting to the server ...',
-      included_types: [],
-      excluded_types: []
-    };
-
-    AnnotationTypes.query(
-      {q: ''},
-      function(responseObject, responseHeaders) {
-        for (var i = 0; i < responseObject.objects.length; ++i) {
-          $scope.annotations.included_types.push(
-            responseObject.objects[i].typename);
-        }
-        $scope.annotations.query_message = '';
-      },
-      function(responseObject, responseHeaders) {
-        $log.error('Query errored with: ' + responseObject);
-        $scope.annotations.query_message = 'Connection to server failed';
-      }
-    );
+.controller('DownloadCtrl', ['$window', '$log', 'AnnotationTypes',
+  function DownloadController($window, $log, AnnotationTypes) {
+    var self = this;
+    self.queryStatus = 'Connecting to the server ...';
+    self.fixedTypes = ['experiment', 'sample_name', 'ml_data_source'];
+    self.includedTypes = [];
+    self.excludedTypes = [];
 
     // Remove a type from inclusion list.
-    $scope.del_type = function(index) {
-      current_item = $scope.annotations.included_types[index];
-      $scope.annotations.included_types.splice(index, 1);
-      $scope.annotations.excluded_types.push(current_item);
+    self.delType = function(index) {
+      var currItem = self.includedTypes[index];
+      self.includedTypes.splice(index, 1);
+      self.excludedTypes.push(currItem);
     };
 
     // Add a type back to inclusion list.
-    $scope.add_type = function(index) {
-      current_item = $scope.annotations.excluded_types[index];
-      $scope.annotations.excluded_types.splice(index, 1);
-      $scope.annotations.included_types.push(current_item);
+    self.addType = function(index) {
+      var currItem = self.excludedTypes[index];
+      self.excludedTypes.splice(index, 1);
+      self.includedTypes.push(currItem);
     };
 
     // Handler of "Download" button click event.
-    $scope.start_download = function() {
+    self.startDownload = function() {
       var uri = '/api/v0/sample/get_annotations/?annotation_types=';
-      uri += $scope.annotations.included_types.join();
-      $window.location.href = uri;  // Call compiled downloading API.
+      uri += self.includedTypes.join();
+      $window.location.href = uri;
     };
+
+    // Retrieve selectable annotation types from the server.
+    AnnotationTypes.get({limit: 0},
+      function success(response) {
+        response.objects.forEach(function(element) {
+          self.includedTypes.push(element.typename);
+        });
+        self.queryStatus = '';
+      },
+      function error(errResponse) {
+        var errMessage = errResponse.statusCode + ' ' + errResponse.statusText;
+        $log.error('Failed to get annotation types: ' + errMessage);
+        self.queryStatus = 'Connection to server failed: ' + errMessage;
```

Comment:
Yes, when this kind of error happens, the only thing an user can do is probably refresh the page. I added more details for two reasons:
1) Many users may be familiar with errors such as "404: page not found". 
2) In case our users need technical help from us, they can say that "I got an error 405 blah blah"; otherwise we will have to ask them to check the console and tell us the details.

I can add "Please refresh the page." at the end of `self.queryStatus` to make it a little more user-friendly.